framework/include/controls/ControlInterface.h

```diff
@@ -228,6 +146,18 @@ ControlInterface::getValidInputParameterIteratorsByName(const std::string & name
   if (output.size() == 0)
     mooseError("The controlled parameter, '" << name << "', in " << _ci_name << " was not found.");
 
+  // Error if the size limit was reached
+  if (count != std::numeric_limits<unsigned int>::max() && output.size() != count)
```

Comment:
This is just funky... we shouldn't allow people to pass in a size limit like this... it doesn't make any sense.  I mean... we're not in the 70's writing `strcpy` here...